classes/items/cart/export/Json.php

```diff
@@ -423,18 +446,69 @@ protected function _validateProduct(ProductCore $product, $attributeId)
     }
 
     /**
+     * Checks for allowed carriers by products in cart
+     *
+     * @param   ShopgateCart    $cart
+     * @return array
+     */
+    protected function _getAllowedCarriersByShopgateCart($cart)
+    {
+        $productIds         = array();
+        $allowedCarriers    = array();
+        $shopId             = $this->getPlugin()->getContext()->shop->id;
+
+        foreach ($cart->getItems() as $item) {
+            list($productId, $attributeId) = ShopgateHelper::getProductIdentifiers($item);
+            $productIds[] = $productId;
```

Comment:
This variable is used for the SQL request. Please add an `(int)` cast to be sure to deal with integers